storage/Utils/SnapperConfig.cc

```diff
@@ -88,18 +91,33 @@ SnapperConfig::post_mount()
 
 
 void
-SnapperConfig::post_add_to_etc_fstab()
+SnapperConfig::post_add_to_etc_fstab( EtcFstab & etc_fstab )
 {
     if ( ! sanity_check() )
         return;
 
-    vector<string> args = {
-        "--step", "3",
-        "--root-prefix", get_root_prefix(),
-        "--default-subvolume-name", get_default_subvolume_name()
-    };
+    // Don't call installation-helper as an external program for this step
+    // because this would modify /etc/fstab on the target system while it is
+    // already open and modified by libstorage during the commit phase, so any
+    // changes made to that file by installation-helper are silently
+    // overwritten by libstorage (bsc#1057443).
 
-    installation_helper( args );
+    // Sample fstab entry:
+    //
+    // UUID=5acfd198-963a-4740-a33e-030b7f305d67 /.snapshots btrfs subvol=@/.snapshots 0 0
+
+
+    FstabEntry * entry = new FstabEntry();
+
+    entry->set_device( get_device_name() );
+    entry->set_mount_point( "/" SNAPSHOTS_DIR );
+    entry->set_mount_opts( MountOpts( string( "subvol=" ) + get_snapshots_subvol_name() ) );
```

Comment:
In https://bugzilla.suse.com/show_bug.cgi?id=1030257 it was discussed that mount options should also be copied from the root filesystem (except dropping ro).